client/network/src/behaviour.rs

```diff
@@ -51,8 +51,12 @@ pub struct Behaviour<B: BlockT> {
 	peer_info: peer_info::PeerInfoBehaviour,
 	/// Discovers nodes of the network.
 	discovery: DiscoveryBehaviour,
+	/// Connection limits.
+	connection_limits: libp2p::connection_limits::Behaviour,
 	/// Generic request-response protocols.
 	request_responses: request_responses::RequestResponsesBehaviour,
+	// AutoNAT.
+	auto_nat: autonat::Behaviour,
```

Comment:
Is this intended? It modifies the networking protocols of Substrate/Polkadot. If yes, this should IMO go through an RFC: https://github.com/polkadot-fellows/RFCs/pulls

Reply:
I find it a bit worrisome that modifying the networking protocol would have passed review

Reply:
Could you @melekes elaborate on why this is added in the first place?

Reply:
Thanks 🙏 Will create an RFC today. I've reverted the code to manually confirming the observed address (although it's not safe I believe)

Reply:
1. Kademlia switches its mode to server once it receives a confirmed external address. Of course, we can set mode to server ourselves (via `set_mode` like it is done in tests), but I don't think it's desirable.
2. I've moved logging of new external addresses (`Discovered new external address for our node`) to `FromSwarm::ExternalAddrConfirmed` case, so we don't log all the candidates.
3. AutoNAT makes sure external address are publicly reachable (currently we're confirming all external addresses https://github.com/paritytech/substrate/pull/14429/commits/d464408c81a83bf355758bc9c128cb12626b9ff0)

Reply:
> (although it's not safe, I believe)

It's how it works today.
There's no safety aspect here. We must already consider every peer as potentially malicious. Increasing the chance that peers report accurate information should neither increase not decrease safety. If it does, it means that we don't actually consider every peer as potentially malicious and this should be fixed.

I'm not necessarily against the principle of AutoNAT. What I'm completely against is sneaking changes to the networking protocol in minor version bumps, and I'm quite annoyed that rust-libp2p does that.

Reply:
Note from the rust-libp2p side:

- running `libp2p-autonat` is not required.
- `libp2p-autonat` is not automatically enabled.
- manually confirming the observed addresses is just fine. If I recall correctly there was a command line flag in Substrate where a user could even specify the external address.

Long story short, Substrate should be able to upgrade to `libp2p` `v0.52.0` without the introduction of `libp2p-autonat`.